public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx

```diff
@@ -122,6 +123,17 @@ export function panelMenuBehavior(menu: VizPanelMenu, isRepeat = false) {
             );
           },
         });
+
+        const gridItem = parent.parent;
+        if (gridItem instanceof DashboardGridItem) {
+          moreSubMenu.push({
+            text: t('panel.header-menu.replace-library-panel', `Replace library panel`),
+            onClick: () => {
+              DashboardInteractions.panelMenuItemClicked('replaceLibraryPanel');
+              dashboard.onShowAddLibraryPanelDrawer(gridItem.getRef());
```

Comment:
Looking at this, compared with how we do it for `Create library panel`, I wonder, why do we need to pass the grid item's ref, not just the panel? In general feels odd to me to just move around grid panels, instead of just the panel, but I guess it is because the DashboardGridItem takes care of the lib panel logic.

Just to understand better. Have we tried to replace just the body of the grid item we're converting to the lib panel? I guess if we could do that, we wouldn't need to find layout components and rows and replace the children, we could just replace the body of the targeted grid item.

Reply:
@ivanortegaalba You're right! I tested again and it worked. I recall trying this when I first opened the PR a month ago and had trouble with it, but I might be wrong.

Indeed it works better by just replacing the viz panel body instead of the entire grid item, I refactored it to show that. I also realised that I applied the same logic in creating library panels [here](https://github.com/grafana/grafana/issues/88359) so that can also be simplified, [PR](https://github.com/grafana/grafana/pull/88681) for that also up. Sorry about all the extra complications.

Reply:
Awesome! I love it when something smells, you drop a comment and we go further to improve it even in other places! Amazing work Victor! Thank you!!!!